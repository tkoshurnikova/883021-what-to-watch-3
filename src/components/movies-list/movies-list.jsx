import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import MovieCard from "../movie-card/movie-card.jsx";
import {connect} from "react-redux";

class MoviesList extends PureComponent {
  render() {
    const {films, onCardClick, cardsToShow, activeItem, onActiveItemChange} = this.props;
    const showedFilms = films.slice(0, cardsToShow);

    return (
      <div className="catalog__movies-list">
        {showedFilms.map((film) => (
          <MovieCard
            film={film}
            key={film.title}
            onCardHover={onActiveItemChange}
            onCardHoverOut={onActiveItemChange}
            onCardClick={onCardClick}
            activeCard={activeItem}
          />
        ))}
      </div>
    );
  }
}

MoviesList.propTypes = {
  films: PropTypes.array.isRequired,
  onCardClick: PropTypes.func.isRequired,
  cardsToShow: PropTypes.number.isRequired,
  activeItem: PropTypes.object,
  onActiveItemChange: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  cardsToShow: state.cardsToShow
});

export {MoviesList};
export default connect(mapStateToProps)(MoviesList);
