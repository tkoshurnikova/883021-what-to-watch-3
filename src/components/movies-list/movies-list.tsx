import * as React from "react";
import MovieCard from "../movie-card/movie-card";
import {connect} from "react-redux";
import {getCardsToShow} from "../../reducer/app/selectors";
import {Operation as DataOperations} from "../../reducer/data/data";
import {Film} from "../../types";

interface Props {
  films: Film[];
  cardsToShow: number;
  activeItem?: Film;
  onActiveItemChange: () => void;
  loadReviews?: (film: {}) => void;
}

const MoviesList: React.FunctionComponent<Props> = (props: Props) => {
  const {films, cardsToShow, activeItem, onActiveItemChange, loadReviews} = props;
  const showedFilms = films.slice(0, cardsToShow);

  return (
    <div className="catalog__movies-list">
      {showedFilms.map((film) => (
        <MovieCard
          film={film}
          key={film.id}
          onCardHover={onActiveItemChange}
          onCardHoverOut={onActiveItemChange}
          activeCard={activeItem}
          loadReviews={loadReviews}
        />
      ))}
    </div>
  );
};

const mapStateToProps = (state) => ({
  cardsToShow: getCardsToShow(state)
});

const mapDispatchToProps = (dispatch) => ({
  loadReviews(item) {
    dispatch(DataOperations.loadReviews(item));
  }
});

export {MoviesList};
export default connect(mapStateToProps, mapDispatchToProps)(MoviesList);
