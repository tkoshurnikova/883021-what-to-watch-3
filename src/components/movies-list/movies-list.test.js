import React from "react";
import renderer from "react-test-renderer";
import {MoviesList} from "./movies-list.jsx";
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

const films = [
  {
    title: `The No Flowers`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Rings of Cloud`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `The Sword's Memory`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `The Voyagers of the Words`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Shores in the Heart`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Missing Illusion`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `The Unwilling Man`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    title: `Teacher in the Hunter`,
    image: `http://placehold.it/280x175`,
    description: `qwerty`,
    rating: 1.0,
    numberOfVotes: 1982,
    director: `Taika Di Caprio`,
    actors: `Ashley Cooper`,
    genre: `Dramas`,
    year: 1900,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

const mockStore = configureMockStore([]);
let store = mockStore({
  filteredFilms: [],
  films,
  genre: ``,
  cardsToShow: 8
});

it(`Render MoviesList`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <MoviesList
            films={films}
            onCardClick={() => {}}
            cardsToShow={8}
            activeItem={films[0]}
            onActiveItemChange={() => {}}
          />)
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
