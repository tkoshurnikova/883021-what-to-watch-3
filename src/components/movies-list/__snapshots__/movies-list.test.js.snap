// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="The No Flowers"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          The No Flowers
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Rings of Cloud"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          Rings of Cloud
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="The Sword's Memory"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          The Sword's Memory
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="The Voyagers of the Words"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          The Voyagers of the Words
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Shores in the Heart"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          Shores in the Heart
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Missing Illusion"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          Missing Illusion
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="The Unwilling Man"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          The Unwilling Man
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Teacher in the Hunter"
          height="175"
          src="http://placehold.it/280x175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page"
          onClick={[Function]}
        >
          Teacher in the Hunter
        </a>
      </h3>
    </article>
  </div>,
  ")",
]
`;
